src/include/miopen/solver/conv_direct_naive_conv.hpp

```diff
@@ -72,7 +72,7 @@ void printTensorStrides(const TensorDescriptor& inDesc,
 template <typename T, unsigned N>
 class MyArray
 {
-    T data_[N];
+    T data_[N] = {0};
```

Comment:
Unfortunately it's a bit confusing initialization of `data_`. And unfortunately there is no way to do simple initialization of such kind of arrays.

It works well in terms of it initialized `data_` with all zeroes, but for example the following code
```C++
T data_[N] = {1};
```
will do the following thing: `data_ = {1, 0, ... ,0};`, where `0` is default-initialization and `1` is value initialization, instead of potentially expected `data_ = {1, 1, ... ,1};`, thought it may perfectly work for `0` - the first element explicitly initialized to `0` and the rest implicitly. But it won't be so simple - the first element is value-initialized and the rest are default-initialized which potentially may be different from zero (for non-simple numerict types).

I just want to clarify this behavior. For me it's better to call fully-implicit initialization like `T data_ = {};` which tells that we want to uniformly default-initialize everything.

Reply:
fixed in latest commit. I was under the impression that `{1}` initialized all to one, but thanks for clarifying. Has to be one of those weird issues with `{}` initialization.